winlogbeat/eventlog/eventlog.go

```diff
@@ -80,53 +82,66 @@ type Record struct {
 
 // ToMapStr returns a new MapStr containing the data from this Record.
 func (e Record) ToEvent() beat.Event {
-	m := common.MapStr{
-		"type":          e.API,
-		"log_name":      e.Channel,
-		"source_name":   e.Provider.Name,
-		"computer_name": e.Computer,
-		"record_number": strconv.FormatUint(e.RecordID, 10),
+	// Windows Log Specific data
+	win := common.MapStr{
+		"channel":       e.Channel,
 		"event_id":      e.EventIdentifier.ID,
+		"provider_name": e.Provider.Name,
+		"record_id":     e.RecordID,
+		"task":          e.Task,
+		"api":           e.API,
 	}
-
-	addOptional(m, "xml", e.XML)
-	addOptional(m, "provider_guid", e.Provider.GUID)
-	addOptional(m, "version", e.Version)
-	addOptional(m, "level", e.Level)
-	addOptional(m, "task", e.Task)
-	addOptional(m, "opcode", e.Opcode)
-	addOptional(m, "keywords", e.Keywords)
-	addOptional(m, "message", sys.RemoveWindowsLineEndings(e.Message))
-	addOptional(m, "message_error", e.RenderErr)
-
+	addOptional(win, "computer_name", e.Computer)
+	addOptional(win, "kernel_time", e.Execution.KernelTime)
+	addOptional(win, "keywords", e.Keywords)
+	addOptional(win, "opcode", e.Opcode)
+	addOptional(win, "processor_id", e.Execution.ProcessorID)
+	addOptional(win, "processor_time", e.Execution.ProcessorTime)
+	addOptional(win, "provider_guid", e.Provider.GUID)
+	addOptional(win, "session_id", e.Execution.SessionID)
+	addOptional(win, "task", e.Task)
+	addOptional(win, "user_time", e.Execution.UserTime)
+	addOptional(win, "version", e.Version)
+	addOptional(win, "message_error", e.RenderErr)
 	// Correlation
-	addOptional(m, "activity_id", e.Correlation.ActivityID)
-	addOptional(m, "related_activity_id", e.Correlation.RelatedActivityID)
-
+	addOptional(win, "activity_id", e.Correlation.ActivityID)
+	addOptional(win, "related_activity_id", e.Correlation.RelatedActivityID)
 	// Execution
-	addOptional(m, "process_id", e.Execution.ProcessID)
-	addOptional(m, "thread_id", e.Execution.ThreadID)
-	addOptional(m, "processor_id", e.Execution.ProcessorID)
-	addOptional(m, "session_id", e.Execution.SessionID)
-	addOptional(m, "kernel_time", e.Execution.KernelTime)
-	addOptional(m, "user_time", e.Execution.UserTime)
-	addOptional(m, "processor_time", e.Execution.ProcessorTime)
+	addOptional(win, "process.pid", e.Execution.ProcessID)
```

Comment:
Because that's not the "userland" PID, this is always the PID of the Windows service processing log events.

The interesting `process.pid` is at event_data.ProcessId, and we've decided not to dig in there right from Go / from the deployed agent.